### ISSUE TITLE
[build] error: can't copy './yang-models/sonic_yang_tree': doesn't exist or not a regular file

### ISSUE THREAD

**Issue by tim-rj** [Azure/sonic-buildimage#7076]
#### Steps to reproduce the issue:
1. build local for 202012 branch
2. make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl build failed
3.

#### Describe the results you received:
```
copying ./yang-models/sonic-port.yang -> build/bdist.linux-x86_64/wheel/sonic_yang_models-1.0.data/data/yang-models
copying ./yang-models/sonic-portchannel.yang -> build/bdist.linux-x86_64/wheel/sonic_yang_models-1.0.data/data/yang-models
copying ./yang-models/sonic-types.yang -> build/bdist.linux-x86_64/wheel/sonic_yang_models-1.0.data/data/yang-models
copying ./yang-models/sonic-versions.yang -> build/bdist.linux-x86_64/wheel/sonic_yang_models-1.0.data/data/yang-models
copying ./yang-models/sonic-vlan.yang -> build/bdist.linux-x86_64/wheel/sonic_yang_models-1.0.data/data/yang-models
error: can't copy './yang-models/sonic_yang_tree': doesn't exist or not a regular file
[  FAIL LOG END  ] [ target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl ]
make: *** [slave.mk:601: target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl] Error 1
make[1]: *** [Makefile.work:281: target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl] Error 2
```

#### Describe the results you expected:
make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl build successfully

#### Output of `show version`:

```
(paste your output here)
```

**Comment by anshuv-mfst** [comment]
@praveen-li - could you please look into this issue, thanks.

**Comment by praveen-li** [comment]
ack

**Comment by praveen-li** [comment]
@tim-rj 

-- build target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl with KEEP_SLAVE_ON=yes i.e. 

make KEEP_SLAVE_ON=yes  target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl

-- you will be in the sonic slave docker when build fails.

-- run below command in src/sonic-yang-models folder, send me the output of this command
pyang -f tree ./yang-models/*.yang > ./yang-models/sonic_yang_tree


OR 
send me full logs of the build :)

**Comment by Sam-Njine** [comment]
I'm getting this error when building the .bin for sonic-broadcom

tests/test_sonic_yang_models.py .F                                       [ 66%]
tests/yang_model_tests/test_yang_model.py .                              [100%]

=================================== FAILURES ===================================
___________________________ test_generate_yang_tree ____________________________

    def test_generate_yang_tree():
    
        # Generate YANG Tree, see no error in it.
        pyang_tree_cmd = "pyang -f tree ./yang-models/*.yang > ./yang-models/sonic_yang_tree"
        if (system(pyang_tree_cmd)):
            print("Failed: {}".format(pyang_tree_cmd))
>           exit(1)
E           SystemExit: 1

tests/test_sonic_yang_models.py:30: SystemExit
----------------------------- Captured stdout call -----------------------------
Failed: pyang -f tree ./yang-models/*.yang > ./yang-models/sonic_yang_tree
----------------------------- Captured stderr call -----------------------------
./yang-models/sonic-acl.yang:8: error: module "ietf-yang-types" not found in search path
./yang-models/sonic-acl.yang:8: warning: imported module "ietf-yang-types" not used
./yang-models/sonic-acl.yang:15: error: module "ietf-inet-types" not found in search path
./yang-models/sonic-acl.yang:27: warning: imported module "sonic-extension" not used
./yang-models/sonic-breakout_cfg.yang:8: warning: imported module "sonic-extension" not used
./yang-models/sonic-device_neighbor.yang:12: warning: imported module "sonic-extension" not used
./yang-models/sonic-interface.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-loopback-interface.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-port.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-portchannel.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-vlan.yang:17: warning: imported module "sonic-extension" not used
./yang-models/sonic-vrf.yang:5: warning: imported module "sonic-extension" not used
====================== 1 failed, 2 passed in 0.91 seconds ======================
[  FAIL LOG END  ] [ target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl ]
make: *** [slave.mk:603: target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl] Error 1

**Comment by Sam-Njine** [comment]
i was able to resolve my issue by building the wheel file from within the sonic slave docker and then copying the .whl file to the target/python-wheels/ folder

**Comment by gck-04** [comment]
> i was able to resolve my issue by building the wheel file from within the sonic slave docker and then copying the .whl file to the target/python-wheels/ folder

Hi Sam,

Could you elaborate the steps/commands you executed to resolve this issue.
Thank you for the help in advance.

Regards,
Gangadhar

**Comment by Sam-Njine** [comment]
@gck-04 

My error had to do with dependencies like libyang and some others not installing when I just run make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl.

What I did was to be in the docker slave while it built as @praveen-li suggested using make KEEP_SLAVE_ON=yes target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl

I checked the log file in the target/python-wheels folder as it listed the missing dependencies which were .deb files in the target/debs/buster folder. I installed the missing dependencies while in the docker slave, then built the wheel file using the instructions to build a wheel file located here https://github.com/Azure/sonic-utilities/blob/master/README.md (port the steps done here to suite the failing sonic_yang whl file)

I copied the .whl file from the folder it was built at to the target/python-wheels/ folder and continued with the make target/sonic-broadcom.bin. It didn't fail again after that.
